cmd/gateway-gcs.go

```diff
@@ -942,7 +942,13 @@ func (l *gcsGateway) CompleteMultipartUpload(bucket string, key string, uploadID
 
 	var parts []*storage.ObjectHandle
 	for _, uploadedPart := range uploadedParts {
-		parts = append(parts, l.client.Bucket(bucket).Object(gcsMultipartDataName(uploadID, uploadedPart.ETag)))
+		parts = append(parts, l.client.Bucket(bucket).Object(gcsMultipartDataName(uploadID,
+			uploadedPart.PartNumber, uploadedPart.ETag)))
+	}
+
+	// Returns name of the composed object.
+	gcsMultipartComposeName := func(uploadID string, composeNumber int) string {
+		return fmt.Sprintf("%s/tmp/%s/composed-object-%05d", gcsMinioMultipartPathV1, uploadID, composeNumber)
```

Comment:
@harshavardhana  just like in FS we have:
.minio.sys/multipart
.minio.sys/tmp

in GCS we can have:
minio.sys.temp/multipart
minio.sys.temp/tmp

Just like FS/XL, all staging files can be in `minio.sys.temp/tmp` This way contents of .minio.sys/multipart/v1/uploadID is clean and predictable. In future if we have to implement ListObjectParts() it will be simple as we don't have to filter out tmp.